python_modules/dagster/dagster/_grpc/impl.py

```diff
@@ -462,20 +462,22 @@ def get_partition_tags(
 
         # Certain gRPC servers do not have access to the instance, so we only attempt to instantiate
         # the instance when necessary for dynamic partitions: https://github.com/dagster-io/dagster/issues/12440
-        instance = (
+        with (
             DagsterInstance.from_ref(instance_ref)
             if (_partitions_def_contains_dynamic_partitions_def(partitions_def) and instance_ref)
-            else None
-        )
-        partition = partitions_def.get_partition(partition_name, dynamic_partitions_store=instance)
-        with user_code_error_boundary(
-            PartitionExecutionError,
-            lambda: f"Error occurred during the evaluation of the `tags_for_partition` function for partitioned config on job '{job_def.name}'",
-        ):
-            tags = partitioned_config.get_tags_for_partition_key(
-                partition.name, job_name=job_def.name
+            else nullcontext()
+        ) as instance:
+            partition = partitions_def.get_partition(
+                partition_name, dynamic_partitions_store=instance
             )
-            return ExternalPartitionTagsData(name=partition.name, tags=tags)
+            with user_code_error_boundary(
+                PartitionExecutionError,
+                lambda: f"Error occurred during the evaluation of the `tags_for_partition` function for partitioned config on job '{job_def.name}'",
+            ):
+                tags = partitioned_config.get_tags_for_partition_key(
+                    partition.name, job_name=job_def.name
```

Comment:
is the fix for the issue where instance needed to be passed in here going in separately?